hyperspy/axes.py

```diff
@@ -655,6 +720,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(axes_manager=self)
+
+    def _on_view_changed(self):
+        self._update_attributes()
+        self.events.axes_changed.trigger()
```

Comment:
I think it is preferable to introduce new events together with a use case, and make this PR as focused as possible. It is already quite large and difficult to manage, and to be honest, I don't have the time.